packages/tailwindcss/src/ast.ts

```diff
@@ -297,7 +301,14 @@ export function optimizeAst(ast: AstNode[]) {
       for (let child of node.nodes) {
         transform(child, copy.nodes, depth + 1)
       }
-      parent.push(copy)
+      if (
+        copy.nodes.length > 0 ||
+        copy.name === '@layer' ||
+        copy.name === '@charset' ||
+        copy.name === '@custom-media'
```

Comment:
```suggestion
        copy.name === '@custom-media' ||
        copy.name === '@namespace'
```

https://developer.mozilla.org/en-US/docs/Web/CSS/@namespace

Reply:
Good catch 👍